docs/how-to-guides/develop-a-prompty/index.md

```diff
@@ -0,0 +1,374 @@
+  parameters:
+    max_tokens: 128
+    temperature: 0.2
+sample:
+  inputs:
+    first_name: John
+    last_name: Doe
+    question: Who is the most famous person in the world?
+---
+system:
+You are an AI assistant who helps people find information.
+As the assistant, you answer questions briefly, succinctly,
+and in a personable manner using markdown and even add some personal flair with appropriate emojis.
+
+# Safety
+- You **should always** reference factual statements to search results based on [relevant documents]
+- Search results based on [relevant documents] may be incomplete or irrelevant. You do not make assumptions
+# Customer
+You are helping {{first_name}} {{last_name}} to find answers to their questions.
+Use their name to address them in your responses.
+
+user:
+  {{question}}
+```
+Users can override prompty's configuration and LLM parameters when loading prompty.
+Users can also use this format `${env:ENV_NAME}` to pass configuration through environment variables.
+
+#### Override model by dict
+
+```python
+from promptflow.core import Prompty
+
+# Load prompty with dict override
+override_model = {
+    "configuration": {
+        "api_key": "${env:OPENAI_API_KEY}",
+        "base_url": "${env:OPENAI_BASE_URL}",
+    },
+    "parameters": {"max_tokens": 512}
+}
+prompty = Prompty.load(source="path/to/prompty.prompty", model=override_model)
+```
+
+#### Override model by OpenAIModelConfiguration
+
+```python
+from promptflow.core import Prompty, OpenAIModelConfiguration
+
+# Load prompty with dict override
+configuration = OpenAIModelConfiguration(
+    model="gpt-35-turbo",
+    base_url="${env:OPENAI_BASE_URL}",
+    api_key="${env:OPENAI_API_KEY}",
+)
+override_model = {
+    "configuration": configuration,
+    "parameters": {"max_tokens": 512}
+}
+prompty = Prompty.load(source="path/to/prompty.prompty", model=override_model)
+```
+
+:::
+::::
+
+## Execute a prompty
+
+Prompty can be executed in these ways to meet the needs of customers in different scenarios.
+
+### Prompty as a call
+After the customer loads the prompty, the loaded prompty object can be called directly as a function. The return value is the content of the LLM response.
+```python
+from promptflow.core import Prompty
+
+prompty_obj = Prompty.load(source="path/to/prompty.prompty")
+result = prompty_obj(first_name="John", last_name="Doh", question="What is the capital of France?")
+```
+
+### Test prompty
+
+#### Test as a flow
+
+::::{tab-set}
+:::{tab-item} CLI
+:sync: CLI
+
+```bash
+# Test prompty with default inputs
+pf flow test --flow path/to/prompty.prompty
+
+# Test prompty with specified inputs
+pf flow test --flow path/to/prompty.prompty --inputs first_name=John last_name=Doh question="What is the capital of France?"
+
+# Test prompty with sample file
+pf flow test --flow path/to/prompty.prompty --inputs path/to/sample.json
+```
+There will be a trace link shown in terminal to visualize the internal execution details for this run.
```

Comment:
```suggestion
There will be a trace link shown in terminal to visualize the internal execution details for this command.
```

Reply:
not sure run is proper here, as above command is a flow test

Reply:
refine it